core/editable.js

```diff
@@ -534,19 +534,28 @@
 			 * Detaches this editable object from the DOM (removes classes, listeners, etc.)
 			 */
 			detach: function() {
-				// Cleanup the element.
-				this.removeClass( 'cke_editable' );
-
 				this.status = 'detached';
 
-				// Save the editor reference which will be lost after
-				// calling detach from super class.
-				var editor = this.editor;
+				// Update the editor cached data with current data.
+				this.editor.setData( this.editor.getData(), 0, 1 );
```

Comment:
Shouldn't `getData` be marked as internal one here?

Reply:
It sounds reasonable